api/services/BedCountService.js

```diff
@@ -0,0 +1,94 @@
+/* global Event Movement */
+'use strict';
+
+var DateRange = require('../lib/DateRange');
+
+// const fullRange = (visibilityRange, rangeFactory) =>
+//   new DateRange(rangeFactory(visibilityRange.from).from, rangeFactory(visibilityRange.to).to);
+
+const fullRange = (visibilityRange, rangeFactory) =>
+  DateRange.widen.apply(null, [visibilityRange, rangeFactory(visibilityRange.from), rangeFactory(visibilityRange.to)]);
+
+let populate = (barrel, centreId, range) =>
+  barrel.find({
+    where: {
+      centre: centreId,
+      timestamp: { '>=': range.from, '<=': range.to }
+    },
+    sort: 'timestamp'
+  });
+
+const populateEvents = (centre, range) =>
+  populate(Event, centre.id, range)
+    .populate('detainee')
+    .then(events => centre.unreconciledEvents = events);
+
+const populateMovements = (centre, range) =>
+  populate(Movement, centre.id, range)
+    .then(movements => centre.unreconciledMovements = movements);
+
+let reconcile = (eventKey, movementKey, centre, reconciler) => {
+  const event = centre.unreconciledEvents[eventKey];
+  const movement = centre.unreconciledMovements[movementKey];
+  if (reconciler(event, movement)) {
+    centre.reconciled.push({ event, movement });
+    delete centre.unreconciledEvents[eventKey];
+    delete centre.unreconciledMovements[movementKey];
+    return true;
+  }
+  return false;
+};
+
+const reconcileEvents = (centre, reconciler) => {
+  centre.unreconciledEvents.forEach((event, eventKey) =>
+    centre.unreconciledMovements.some((movement, movementKey) => {
+      return reconcile(eventKey, movementKey, centre, reconciler);
+    })
+  );
+  centre.unreconciledMovements = centre.unreconciledMovements.filter((movement) => movement !== null);
+  centre.unreconciledEvents = centre.unreconciledEvents.filter((event) => event !== null);
+};
+
+const resolveEventOperationWithMovementDirection = (eventType) => {
+  switch (eventType) {
+  case 'check in':
+    return 'in';
+  case 'check out':
+    return 'out';
+  default:
+    return 'unknown';
```

Comment:
The code actually wants to pass gracefully over events with operations which aren't check in or check out